crates/ide-assists/src/handlers/raw_string.rs

```diff
@@ -155,7 +155,7 @@ pub(crate) fn remove_hash(acc: &mut Assists, ctx: &AssistContext<'_>) -> Option<
     })
 }
 
-fn required_hashes(s: &str) -> usize {
+pub(crate) fn required_hashes(s: &str) -> usize {
```

Comment:
This function sounds general purpose enough that I think we can move this into `src/utils.rs`